src/combinatorics/simple_graph/triangle/basic.lean

```diff
@@ -0,0 +1,62 @@
+/-
+Copyright (c) 2022 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import combinatorics.simple_graph.clique
+
+/-!
+# Triangles in graphs
+
+This file defines triangles in simple graphs.
+
+A triangle is a `3`-clique, that is a set of three vertices which are all connected.
+
+## Main declarations
+
+* `simple_graph.triangle_free_far`: Predicate for a graph to have enough triangles that one must
+  remove a lot of edges to remove all of them. This is the crux of the Triangle Removal lemma.
```

Comment:
```suggestion
  remove a lot of edges to remove all of them. This is the crux of the Triangle Removal lemma.
  
  ## TODO
  
  * Generalise `triangle_free_far` to other graphs, to state and prove the Graph Removal Lemma.
```

(the reason I don't want to do this in this PR is that I *think* the definition should change to `epsilon n^(|H|-1)` for a graph `H`, but I'm not certain enough yet to want that in mathlib)